src/threading.ts

```diff
@@ -117,12 +118,24 @@ class Thread {
     public writeTraceback(): void {
         this.traceback = "Traceback (most recent call last):\n" + this.traceback;
         process.stdout.write(this.traceback);
+        this.exit();
     }
 
-    // Output the traceback when thread cannot handle an exception, and exit the thread
-    public throwException(): void {
-        this.writeTraceback();
-        this.exit();
+    public throwException(exc: IPy_Object): void {
+        // Whenever an exception occurs, tries to find a handler and if it can't outputs the traceback 
+        this.exc = exc;
+        var f: IPy_FrameObj = this.getTopOfStack();
+        if (!f.tryCatchException(this)) {
+            // Find in previous frames
+            for (var i = this.stack.length - 2; i >= 0; i--) {
```

Comment:
Whoops. I figured out what I was doing wrong. The exceptions need to be pushed on to the calling function's frame stack, if the current frame cannot handle the raised exception.
```python
def foo():
    asdf

def nested():
    try:
        foo()
    except Exception:
        print "Success"
```
So in this case, ```foo``` needs to push the ```NameError``` exception onto the ```nested's``` frame stack.